ios-offline-maps/ios-offline-maps/TileRegionDownloadView.swift

```diff
@@ -0,0 +1,252 @@
+    
+    // define regions
+    private let regions = [
+        OfflineRegion(
+            id: "new-york-region",
+            name: "New York",
+            bounds: CoordinateBounds(
+                southwest: CLLocationCoordinate2D(
+                  latitude:  40.48398,
+                  longitude: -74.28127
+                ),
+                northeast: CLLocationCoordinate2D(
+                  latitude: 40.98701,
+                  longitude: -73.58442
+                )
+              )
+        ),
+        OfflineRegion(
+            id: "london-region",
+            name: "London",
+            bounds: CoordinateBounds(
+                southwest: CLLocationCoordinate2D(latitude: 51.4874, longitude: -0.1278),
+                northeast: CLLocationCoordinate2D(latitude: 51.5174, longitude: -0.0978)
+            )
+        ),
+        OfflineRegion(
+            id: "paris-region",
+            name: "Paris",
+            bounds: CoordinateBounds(
+                southwest: CLLocationCoordinate2D(latitude: 48.8366, longitude: 2.3522),
+                northeast: CLLocationCoordinate2D(latitude: 48.8666, longitude: 2.3822)
+            )
+        )
+    ]
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                // Header
+                HStack {
+                    Button(action: {
+                        dismiss()
+                    }) {
+                        HStack {
+                            Image(systemName: "chevron.left")
+                            Text("Back")
+                        }
+                        .foregroundColor(.blue)
+                    }
+                    Spacer()
+                    Text("Offline Regions")
+                        .font(.title2)
+                        .fontWeight(.semibold)
+                    Spacer()
+                    Button(action: {
+                        clearAllRegions()
+                    }) {
+                        Text("Clear All")
+                            .font(.caption)
+                            .foregroundColor(.red)
+                    }
+                }
+                .padding()
+                
+                // List of regions
+                List {
+                    ForEach(regions, id: \.id) { region in
+                        RegionRowView(
+                            regionId: region.id,
+                            regionName: region.name,
+                            isDownloading: downloadingRegions.contains(region.id),
+                            progress: downloadProgress[region.id] ?? 0.0,
+                            refreshTrigger: refreshTrigger,
+                            onDownload: {
+                                downloadRegion(region: region)
+                            }
+                        )
+                    }
+                }
+            }
+        }
+        .navigationBarHidden(true)
```

Comment:
```suggestion
            .navigationTitle("Offline Regions")
            .toolbarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .topBarLeading) {
                    Button("Done", role: .cancel) {
                        dismiss()
                    }
                }
                ToolbarItem(placement: .topBarTrailing) {
                    Button("Clear All", role: .destructive) {
                        clearAllRegions()
                    }
                    .tint(.red)
                }
            }
```

I suggest using native nav bar for the look and feel